node/src/blockchain/blockchain_interface.rs

```diff
@@ -799,14 +883,18 @@ mod tests {
             .into_iter()
             .map(|request| serde_json::from_slice(&request.body()).unwrap())
             .collect();
+        let bodies: Vec<String> = bodies
+            .into_iter()
+            .map(|b| serde_json::to_string(&b).unwrap())
+            .collect();
         assert_eq!(
-            format!("\"0x000000000000000000000000{}\"", &to[2..]),
-            bodies[0]["params"][0]["topics"][2].to_string(),
+            vec![format!("[{{\"id\":0,\"jsonrpc\":\"2.0\",\"method\":\"eth_blockNumber\",\"params\":[]}},{{\"id\":1,\"jsonrpc\":\"2.0\",\"method\":\"eth_getLogs\",\"params\":[{{\"address\":\"0x384dec25e03f94931767ce4c3556168468ba24c3\",\"fromBlock\":\"0x2a\",\"toBlock\":\"0x400\",\"topics\":[\"0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef\",null,\"0x000000000000000000000000{}\"]}}]}}]", &to[2..])],
+            bodies,
         );
         assert_eq!(
             result,
             RetrievedBlockchainTransactions {
-                new_start_block: 0x4be663 + 1,
+                new_start_block: 0x4be663,
```

Comment:
Is this block going to be included in the next scan? Because if yes then it is the same as the block of the last transaction we found and so we would pick that transaction again. Please verify.